addons/mail/static/tests/chatter_tests.js

```diff
@@ -291,6 +302,79 @@ QUnit.test('Activity Done keep feedback on blur', function (assert) {
     });
 });
 
+QUnit.test('Activity Done by uploading a file', function (assert) {
+    assert.expect(4);
+
+    // simulate (shortcut) the upload and trigger the event when the
+    // attachment is created serverside.
+    testUtils.mock.patch(Activity, {
+        _onMarkActivityDoneUploadFile: function (ev) {
+            var $markDoneBtn = $(ev.currentTarget);
+            var fileuploadID = $markDoneBtn.data('fileupload-id');
+
+            var win = window.top.window;
+            win.jQuery(win).trigger(fileuploadID, [{
```

Comment:
why not simply `$(window).trigger()`? Maybe there's a reason you didn't do that?

Reply:
I just copy/paste the code returned by the upload_file controller. I will change this.